### PR TITLE
Change client/server version support

### PR DESCRIPTION
**Pull request by alexshtin**
Wire backward compatible changes: 
1. Rename `ClientVersionNotSupportedFailure.client_impl` -> `ClientVersionNotSupportedFailure.client_name`.
2. Remove `FeatureVersionNotSupportedFailure`.
3. Add `ServerVersionNotSupportedFailure`.
4. Remove `RespondQueryTaskCompletedRequest.worker_version_info`.

Wire breaking change:
1. `GetClusterInfoResponse` changed completely but `GetClusterInfo` API shouldn't be used by anyone.

### CODE REVIEW

**Review by samarabbas — APPROVED** [review]

File: temporal/api/errordetails/v1/message.proto [hunk 1]
```diff
@@ -50,14 +50,13 @@ message NamespaceNotActiveFailure {
...
     string supported_versions = 3;
 }
 
-message FeatureVersionNotSupportedFailure {
-    string feature = 1;
-    string feature_version = 2;
-    string supported_versions = 3;
+message ServerVersionNotSupportedFailure {
+    string server_version = 1;
+    string supported_versions = 2;
```
Comment (mfateev) [inline temporal/api/errordetails/v1/message.proto:59]: Maybe make it a repeated field?

File: temporal/api/errordetails/v1/message.proto [hunk 2]
```diff
@@ -50,14 +50,13 @@ message NamespaceNotActiveFailure {
 
 message ClientVersionNotSupportedFailure {
     string client_version = 1;
-    string client_impl = 2;
+    string client_name = 2;
```
Comment (mfateev) [inline temporal/api/errordetails/v1/message.proto:53]: Maybe make it a repeated field?

File: temporal/api/workflowservice/v1/request_response.proto [hunk 3]
```diff
@@ -568,9 +567,10 @@ message DescribeTaskQueueResponse {
 message GetClusterInfoRequest {
 }
 
-// GetClusterInfoResponse contains information about Temporal cluster
+// GetClusterInfoResponse contains information about Temporal cluster.
 message GetClusterInfoResponse {
-    temporal.api.version.v1.SupportedSDKVersions supported_sdk_versions = 1;
+    map<string,string> supported_clients = 1;
```
Comment (mfateev) [inline temporal/api/workflowservice/v1/request_response.proto:572]: Add comments about this map key and value formats.

File: temporal/api/errordetails/v1/message.proto (hunk 1 above)
Comment (alexshtin) [reply temporal/api/errordetails/v1/message.proto:59]: The library I am using supports multiple ranges in one query: `<1.0.0 || >1.5.0  <2.0.0 || >2.5.0 <5.0.0`. So I expect it to be a single string even if we have many ranges.

Comment (alexshtin) [reply temporal/api/errordetails/v1/message.proto:59]: https://github.com/blang/semver#ranges

**Review by mfateev — APPROVED** [review]

**Pull request merged**